mbf_costmap_nav/src/mbf_costmap_nav/costmap_navigation_server.cpp

```diff
@@ -825,4 +816,82 @@ bool CostmapNavigationServer::callServiceClearCostmaps(std_srvs::Empty::Request
   return true;
 }
 
+CostmapWrapper::Ptr CostmapNavigationServer::selectingRequestedCostmap(std::uint8_t costmap_type) const
+{
+  // selecting the requested costmap
+  CostmapWrapper::Ptr costmap;
+  switch (costmap_type)
+  {
+    case mbf_msgs::CheckPose::Request::LOCAL_COSTMAP:
+      return local_costmap_ptr_;
+      break;
+    case mbf_msgs::CheckPose::Request::GLOBAL_COSTMAP:
+      return global_costmap_ptr_;
+      break;
+    default:
+      ROS_ERROR_STREAM("No valid costmap provided; options are "
+                       << mbf_msgs::CheckPose::Request::LOCAL_COSTMAP << ": local costmap, "
+                       << mbf_msgs::CheckPose::Request::GLOBAL_COSTMAP << ": global costmap");
+      return costmap;
+  }
+}
+
+bool CostmapNavigationServer::callServiceFindValidPose(mbf_msgs::FindValidPose::Request& request,
+                                                       mbf_msgs::FindValidPose::Response& response)
+{
+  CostmapWrapper::Ptr costmap = selectingRequestedCostmap(request.costmap);
+  if (!costmap)
+  {
+    return false;
+  }
+  std::string costmap_name =
```

Comment:
can be declared const. Not sure, if you want to do that good practice in general.